### DISCUSSION TITLE
Test Push Notifications on iOS Simulator

### DISCUSSION THREAD

**Discussion by jrmccann2** [invertase/react-native-firebase#7509] [Q&A] [answered]
I want to test my app's handling of push notifications on an iOS simulator. In the past this was not possible. I see that the documentation now states "You may use a physical iOS device to receive messages or an iOS Simulator if you are on macOS 13+ on Apple Silicon using a Simulator running iOS 16+" (https://rnfirebase.io/messaging/usage/ios-setup). I meet both of those requirements and have been successfully using RN FCM in my app for years. But testing on an iOS simulator is not working. Is there any additional setup that needs to be done to get this working? If so, could we get that added to the documentation? Is there a minimum version of RN FCM that we need to be using? I'm currently on v14.

**Comment by mikehardy** [comment, marked as answer]
> I'm currently on v14

Wow that's old! :-)

Did you try looking at our changelog? https://github.com/invertase/react-native-firebase/blob/main/CHANGELOG.md

I did a brief search for "simulator" and the second instance of the word had your answer (spoiler alert: v17 here is what added it) - careful of the note that this only works on Apple Silicon machines

**Reply by jrmccann2** [reply]
Thanks for the prompt answer. And for not shaming me for skipping a pretty basic step that I should have done... My bad.

And don't worry, we're planning to upgrade soon ;-)

**Reply by Prazitstha** [reply]
@mikehardy Sir, Im on React native 0.64.2 so installing the latest firebase package isnt possible in my case and im using 14.12.0 but im not receiving notifications. Any Solution ?

**Reply by mikehardy** [reply]
> @mikehardy Sir, Im on React native 0.64.2 so installing the latest firebase package isnt possible in my case and im using 14.12.0 but im not receiving notifications. Any Solution ? 
> 

@Prazitstha I don't believe it's valid to base an app on unsupported versions of react-native, and upgrading is the solution as a first step
